nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/BaseNDArrayFactory.java

```diff
@@ -197,7 +197,32 @@ public INDArray linspace(int lower, int upper, int num, DataType dtype) {
         for (int i = 0; i < num; i++) {
             double t = (double) i / (num - 1);
             data[i] = lower * (1 - t) + t * upper;
+        }
+
+        //edge case for scalars
+        INDArray ret = Nd4j.createUninitialized(dtype, new long[]{data.length});
+        if (ret.isScalar())
+            return ret;
```

Comment:
This looks wrong to me... No data is put for scalars - array content (single value) remains uninitialized.